src/route/app.rs

```diff
@@ -405,15 +413,40 @@ impl error::Error for Error {
             }
             Error::InvalidRequestIdHeader(..) => "invalid tracing header value",
             Error::InvalidBodyRead(..) => "failed to read HTTP body",
+            Error::InvalidCocaineHttpProtoVersion => "unsupported X-Cocaine-HTTP-Proto-Version",
             Error::Canceled => "canceled",
         }
     }
 }
 
+/// App can set it by X-Cocaine-HTTP-Proto-Version header
+/// It controls when the proxy determines to close the HTTP request
+#[derive(Copy, Clone, Debug)]
+enum CocaineHttpProtoVersion {
+    /// "1.0"
+    /// Default behavior. Finish HTTP reply when Close arrives from an app
+    Proto10,
+    /// "1.1"
+    /// Finish HTTP reply when Close or empty body arrives from an app
+    Proto11,
+}
+
+impl FromStr for CocaineHttpProtoVersion {
+    type Err = Error;
+    fn from_str(s: &str) -> Result<CocaineHttpProtoVersion, Error> {
+        Ok(match s {
```

Comment:
I try to avoid nested braces.

This looks better:
```rust
match s {
    "1.0" => Ok(Proto10),
    "1.1" => Ok(Proto11),
    _ => Err(Error::InvalidCocaineHttpProtoVersion),
}
```